### PR TITLE
[PE-31] feat: Add lock unlock archive restore realtime sync

### PR DESCRIPTION
**Pull request by Palanikannan1437**
## Description

1. All events regarding Pages (Lock, Unlock, Archive, Restore) are now realtime! 
2. And actions like "favorites" that are personalized to a user are synced on tab re-focus.

## Release Notes

- **New Features**
	- Introduced real-time collaboration capabilities, allowing users to emit and listen for document updates.
	- Enhanced image handling with improved loading and resizing logic in the CustomImageBlock component.
	- Added a new hook for managing collaborative page actions, streamlining state changes for locking, unlocking, archiving, and restoring pages.
	- Implemented a new method for broadcasting client events in the HocusPocus server configuration.
	- Introduced a new constant for collaborative event types, improving document management.
	- Updated the PageDetailsPage component to revalidate data more frequently.
	- Enhanced the PageOptionsDropdown component with collaborative actions for better state management.

- **Bug Fixes**
	- Improved error handling in the SlashCommandsMenu for keyboard navigation.

- **Chores**
	- Expanded type definitions and exports for better modularity and type safety across the codebase.
	- Updated the `IPage` interface to allow for more granular control over synchronization behavior in page state management.

### CODE REVIEW

File: packages/editor/src/core/helpers/document-events.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export const documentEventResponses = {
```
Comment (aaryan610) [inline packages/editor/src/core/helpers/document-events.ts:1]: Convert this to an `enum`

File: packages/editor/src/core/hooks/use-editor.ts [hunk 2]
```diff
@@ -17,6 +17,7 @@ import { IMarking, scrollSummary } from "@/helpers/scroll-to-node";
 import { CoreEditorProps } from "@/props";
 // types
 import { EditorRefApi, IMentionHighlight, IMentionSuggestion, TEditorCommands, TFileHandler } from "@/types";
+import { DocumentEventsServer } from "src/lib";
```
Comment (aaryan610) [inline packages/editor/src/core/hooks/use-editor.ts:20]: Update import statement and order

File: packages/editor/src/core/hooks/use-editor.ts [hunk 3]
```diff
@@ -34,11 +35,11 @@ export interface CustomEditorProps {
   };
   onChange?: (json: object, html: string) => void;
   placeholder?: string | ((isFocused: boolean, value: string) => string);
-  provider?: HocuspocusProvider;
```
Comment (aaryan610) [inline packages/editor/src/core/hooks/use-editor.ts:37]: Update props order

File: packages/editor/src/core/hooks/use-read-only-collaborative-editor.ts [hunk 4]
```diff
@@ -47,23 +47,23 @@ export const useReadOnlyCollaborativeEditor = (props: TReadOnlyCollaborativeEdit
         },
         onSynced: () => setHasServerSynced(true),
       }),
-    [id, realtimeConfig, user]
+    [id, realtimeConfig, serverHandler, user.id]
```
Comment (aaryan610) [inline packages/editor/src/core/hooks/use-read-only-collaborative-editor.ts:50]: Why is the dependecy just `user.id` and not the whole `user` object.

**Review by rahulramesha — APPROVED** [review]

**Pull request merged**